compiler-k2/src/main/kotlin/com/squareup/anvil/compiler/k2/names/DaggerNames.kt

```diff
@@ -0,0 +1,7 @@
+package com.squareup.anvil.compiler.k2.names
+
+import org.jetbrains.kotlin.name.FqName
+
+internal object DaggerNames {
```

Comment:
The `Names` object I created in the big POC branch was intended to aid discoverability with IDE auto-complete, since everything's hierarchal under `Names`.  It also has the benefit of not adding any new top-level types that start with Dagger, Anvil, or Java.

Would you mind continuing with that pattern instead?

Reply:
I noticed that @JoelWilcox  pr also has this DaggerNames object: https://github.com/square/anvil/pull/1112/files#diff-c901a8b204aaf65a92fd66334072a95ae6bbbe36fa2ccf0ce57c65bcb16811f7. So we'll need to align on something across both pr's. My main reason for moving away from the Names.dagger pattern was not wanting to declare objects with lowercase names. I'm open to other options for organizing these names, maybe just one large flat hierarchy of Names.daggerBinds, Names.daggerModule, Names.javaInject etc.

Reply:
I would also like to avoid having lowecase-named objects, but otherwise don't have strong feelings on how we organize these names. The discoverability aspect makes sense to me :+1:

Reply:
I also didn't/don't like creating lowercase objects, but I rationalize it away by telling myself it's a mini DSL, so the normal conventions don't really apply.

But if you guys would like to capitalize the inner objects, I can live with it.

Reply:
Throwing this out as an option too which would still allow a more-DSL-esque style of accessing the different names, maintain easy discoverability, and allow using specific sets directly if you already know what you want:

Names.kt
```
object Names {
  val dagger = DaggerNames
  val otherGroup = OtherGroupNames
  ...
}
```

DaggerNames.kt
```
internal object DaggerNames {
  val factory = FqName("dagger.internal.Factory")
}
```

Reply:
Oh that's an interesting idea, but I think it could lead to inconsistent usages with some call sites accessing DaggerNames.factory directly and some accessing it via Names.dagger.factory. I'm really leaning towards just keeping this really simple with just a single Names object and then having everything prefix with dagger/javax/anvil etc and sorting the vals alphabetically.

Reply:
sounds good to me!